build.gradle

```diff
@@ -119,78 +91,89 @@ task resetAndTest {
+// Defines a configuration for the MarkLogic modules; used by the modulesJar task below
 configurations {
     modules
 }
+
 task sourcesJar(type: Jar) {
-    description = "Create a jar of the modules to meet jcenter's requirements for publishing"
+    description = "Create a jar of the modules to meet Maven Central's requirements for publishing"
     classifier "sources"
     from "src/main/ml-modules"
     destinationDir file("build/libs")
 }
 
+task javadocJar(type: Jar, dependsOn: javadoc) {
+    classifier "javadoc"
+    from javadoc
+}
+
 task modulesJar(type: Jar) {
-    description = "Jar up the marklogic-geo-data-services MarkLogic modules into a package that can be published"
+    description = "Jar up the marklogic-unit-test MarkLogic modules into a package that can be published"
     from("src/main/ml-config") {
         into("marklogic-geo-data-services-modules/ml-config")
     }
-    // Not including ml-data since the user will need to specify their own
     from("src/main/ml-modules") {
         into("marklogic-geo-data-services-modules/ml-modules")
     }
     from("src/main/ml-plugins") {
         into("marklogic-geo-data-services-modules/ml-plugins")
     }
-    // Not including ml-schemas since the user will need to specify their own
     destinationDir file("build/libs")
     baseName "marklogic-geo-data-services-modules"
 }
 
-// Define the artifacts, in addition to the jar registered by the "java" plugin
 artifacts {
-    modules modulesJar
+    archives sourcesJar, javadocJar, modulesJar
 }
-
-// Setup publishing of both a jar (containing the src/main/java code) and the modules zip
-// Can eventually include a test jar (containing some src/test/java code) if desired
-// Run "gradle publishtomavenlocal" to publish these locally
-publishing.publications {
-    sourcesJava(MavenPublication) {
-        groupId "com.marklogic"
-        artifactId "marklogic-geo-data-services-modules"
-        artifact sourcesJar
-    }
-    mainModules(MavenPublication) {
-        groupId "com.marklogic"
-        artifactId "marklogic-geo-data-services-modules"
-        artifact modulesJar
-    }
+signing {
+    sign configurations.archives
 }
 
-if (project.hasProperty("myBintrayUser")) {
-    bintray {
-        println "Publishing to BinTray:"
-        println "myBintrayUser: " + project.myBintrayUser
-        println "myBintrayKey: " + project.myBintrayKey
-        println "name: " + project.name
-        println "version: " + project.version
-
-        user = project.myBintrayUser
-        key = project.myBintrayKey
-        publications = ["sourcesJava", "mainModules"]
-        pkg {
-            repo = "Maven"
-            name = project.name
-            userOrg = "marklogic-community"
-            licenses = ["Apache-2.0"]
-            vcsUrl = "https://github.com/marklogic-community/marklogic-geo-data-services.git"
-            version {
-                name = project.version
-                released = new Date()
+publishing {
```

Comment:
This is the same config as in our other projects and should "just work" when we try to publish to Maven Central. I'm going to look into getting this published in our internal Nexus too.